src/utils.jl

```diff
@@ -180,5 +180,47 @@ makesym(t::Symbolic; kwargs...) = Sym{symtype(t)}(tosymbol(t; kwargs...))
 makesym(t::Num; kwargs...) = makesym(value(t); kwargs...)
 
 var_from_nested_derivative(x, i=0) = (missing, missing)
-var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1],i+1) : (x,i)
-var_from_nested_derivative(x::Sym,i=0) = (x,i)
+var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1], i + 1) : (x, i)
+var_from_nested_derivative(x::Sym,i=0) = (x, i)
+
+function degree(p::Sym, sym=nothing)
+    if isequal(sym, nothing)
+        return 1
+    else
+        return isequal(p, sym) ? 1 : 0
+    end
+end
+
+function degree(p::Pow, sym=nothing)
+    return p.exp * degree(p.base, sym)
+end
+
+function degree(p::Add, sym=nothing)
+    return maximum([degree(key, sym) for key in keys(p.dict)])
+end
+        
+function degree(p::Mul, sym=nothing)
+    if isequal(sym, nothing)
+        return sum(values(p.dict))
+    else
+        return get(p.dict, sym, 0)
+    end
+end
+
+function degree(p::Term, sym=nothing)
+    if sym === nothing
+        return 1
+    else
+        return isequal(p, sym) ? 1 : 0 
```

Comment:
A more elegant way could be
```suggestion
        return Int(isequal(p, sym))
```
It's a matter of taste though.